sources/packages/backend/apps/api/Dockerfile

```diff
@@ -23,5 +23,10 @@ RUN npm run build api
 # Exposing application port
 EXPOSE ${PORT}
 
+# Grant access to group 0 to allow npm v9 to work
+# See: https://docs.openshift.com/container-platform/4.11/openshift_images/create-images.html#use-uid_create-images
+RUN chgrp -R 0 ./.npm && \
```

Comment:
If the problem is in the cache files' permissions, why we don't clean the cache and let it be created with the running (whichever it is) like in https://stackoverflow.com/a/60192811 ?

Reply:
We should not use sudo in our Dockerfile and yes while running cache clean, the arbitrary user that is trying to clean it should have write access to the folder. So as per the suggestions, I have given group write permissions for the group 0, where those users are assigned.

Reply:
The idea is not to use sudo, the idea is to run  `npm cache clean --force`.

Reply:
the error actually happens not on the build process, rather when there is an error and the crash loop of a pod happens and it restarts. SO when the pod tries to run again, the cache folder does not have write permission by the arbitrary users and it fails.

Reply:
@andrepestana-aot first of all, the problem is not just about the cache file permissions. The problem is about the access for the arbitary user(which openshift uses to run the container) to the to the directory `/opt/app-root/src/.npm` and it's child directories and files. 

By default, this directory is owned by the default user(1001) but the container is started with arbitrarily assigned user who does not have write access to the `.npm` directory.

![image](https://github.com/bcgov/SIMS/assets/54600590/067f44b8-d657-4001-88fd-ee55f6f11dd7)

arbitrarily assigned user
![image](https://github.com/bcgov/SIMS/assets/54600590/fa5e64eb-2cf0-4c7e-9772-a30fa538b710)

Hence it would be required to give permission to `.npm` directory to make sure npm life cycle scrips can work without any issue. 

Please let me know if I am missing something or any further clarification needed.

Reply:
I'm asking because the error we see in the ticket refers to the cache:
![image](https://github.com/bcgov/SIMS/assets/78114138/512bdf8f-8614-48bc-9dda-c78a9b5bb085)

Did we have that error for any other file outside _cacache dir?

Reply:
Yes, in the same screenshot we have the same issue for _logs directory as well.

Reply:
Hm, I see. I did some test to check if the `npm cache clean --force` would delete the _logs but it doesn't. It does remove the _cacache.